models/context.go

```diff
@@ -0,0 +1,73 @@
+// Copyright 2019 The Gitea Authors. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+package models
+
+import "xorm.io/builder"
+
+// DBContext represents a db context
+type DBContext struct {
+	e Engine
+}
+
+// Insert inserts a object to database
+func (ctx *DBContext) Insert(obj interface{}) error {
+	_, err := ctx.e.Insert(obj)
+	return err
+}
+
+// LoadByID loads record from database according id, if it's not exist return ErrNotExist
+func (ctx *DBContext) LoadByID(id int64, obj interface{}) error {
```

Comment:
I don't think this kind of interaction should be exposed outside models. I'd prefer to create a by-pass function when needed, like:
```
type DBContext struct {
	e Engine
}
func UpdateIssueMentionsCtx(ctx DBContext, issueID int64, mentions []string) error {
    return UpdateIssueMentions(ctx.Engine, issueID, mentions)
}
func UpdateIssueMentions(e Engine, issueID int64, mentions []string) error {
    [...]
}
```

Reply:
I will follow that idea.

Reply:
@guillep2k done.